omc3/plotting/utils/windows.py

```diff
@@ -0,0 +1,208 @@
+    )
+
+
+class PlotWidget(QWidget):
+
+    def __init__(self, *figures: Figure, title: str):
+        """A derived QWidget that contains Matplotlib-Figures,
+        stacked vertically, each with its own toolbar.
+
+        Args:
+            *figures (Figure): Figures to be contained in the widget
+            title (str): Name of the widget
+            skip_super (bool): If True the superclass __init__ will be skipped (ONLY FOR TESTS)
+        """
+        QWidget.__init__(self)   # no super(), because will not work with tests
+
+        self.title: str = title
+        self.figures: Tuple[Figure] = figures
+        self._canvas_toolbar_handles: List[Tuple[FigureCanvas, NavigationToolbar]] = []
+
+        self._layout = QVBoxLayout()
+        self.setLayout(self._layout)
+
+        for figure in figures:
+            canvas = FigureCanvas(figure)
+            toolbar = NavigationToolbar(canvas, self)
+
+            self._layout.addWidget(canvas)
+            self._layout.addWidget(toolbar)
+            self._canvas_toolbar_handles.append((canvas, toolbar))
+
+
+class TabWidget(QTabWidget):
+
+    def __init__(self, title: str = None):
+        """A simple tab widget, that in addition
+        to QTabWidget keeps track of the tabs via dictionary
+        and can have a title.
+        
+        Args:
+            title (str): Title of the created Window
+            skip_super (bool): If True the superclass __init__ will be skipped (ONLY FOR TESTS)
+        """
+        QTabWidget.__init__(self)  # no super(), because will not work with tests
+
+        self.title: str = title
+        self.tabs: Dict[str, QWidget] = {}
+
+    def add_tab(self, widget: QWidget):
+        self.tabs[widget.title] = widget
+        self.addTab(widget, widget.title)
+
+
+class SimpleTabWindow:
+
+    def __init__(self, title: str = "Simple Tab Window", size: Tuple[int, int] = None):
+        """A simple window that contains a single Tab-Widget,
+        allowing the user to add tabs to it.
+
+        Args:
+            title (str): Title of the created Window
+            size (Tuple[int, int]): Size of the created window in pixels.
+        """
+        self.app = QApplication(sys.argv)
+        self.main_window = QMainWindow()
+        self.main_window.__init__()
+        self.main_window.setWindowTitle(title)
+        self.current_window = -1
+        self.tabs_widget = TabWidget()
+        self.main_window.setCentralWidget(self.tabs_widget)
+        
+        if size is None:
+            size_in_inch = rcParams["figure.figsize"]
+            size = (int(size_in_inch[0] * 100), int(size_in_inch[1] * 100))  # Approximate
+        self.main_window.resize(*size)
+
+        self.main_window.show()
+
+    def add_tab(self, widget: QWidget):
+        """ Add a new tab made from the given widget.
+
+        Args:
+            widget (QWidget): Widget to add.
+        """
+        self.tabs_widget.add_tab(widget)
+
+    def show(self):
+        self.app.exec_()
```

Comment:
Also related to Qt5 being EOL, if we want to go for Qt6 this would be `.exec()` then.